osin/v1/types.go

```diff
@@ -395,19 +395,19 @@ type TokenConfig struct {
 	AuthorizeTokenMaxAgeSeconds int32 `json:"authorizeTokenMaxAgeSeconds"`
 	// accessTokenMaxAgeSeconds defines the maximum age of access tokens
 	AccessTokenMaxAgeSeconds int32 `json:"accessTokenMaxAgeSeconds"`
-	// accessTokenInactivityTimeoutSeconds defined the default token
-	// inactivity timeout for tokens granted by any client.
-	// Setting it to nil means the feature is completely disabled (default)
-	// The default setting can be overriden on OAuthClient basis.
+	// accessTokenInactivityTimeoutSeconds - DEPRECATED: setting this field has no effect.
+	// +optional
+	AccessTokenInactivityTimeoutSeconds *int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
+	// accessTokenInactivityTimeout defines the token inactivity timeout
+	// for tokens granted by any client.
 	// The value represents the maximum amount of time that can occur between
 	// consecutive uses of the token. Tokens become invalid if they are not
 	// used within this temporal window. The user will need to acquire a new
 	// token to regain access once a token times out.
-	// Valid values are:
-	// - 0: Tokens never time out
-	// - X: Tokens time out if there is no activity for X seconds
-	// The current minimum allowed value for X is 300 (5 minutes)
-	AccessTokenInactivityTimeoutSeconds *int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
+	// If this value is not set, then tokens are valid until their expiry.
+	// Takes valid time duration string such as "5m", "1.5h" or "2h45m".
+	// +optional
+	AccessTokenInactivityTimeout metav1.Duration `json:"accessTokenInactivityTimeout,omitempty"`
```

Comment:
if we are to do this, we should probably keep this field to be of the same type for `osin` and `kubecontrolplane` - https://github.com/openshift/library-go/pull/816#discussion_r446132633

Reply:
Inherently they are the same type as `KubeAPIServerConfig` [re-uses](https://github.com/openshift/api/blob/master/kubecontrolplane/v1/types.go#L58) `osin` config.

Reply:
Good, so we only need it to be a pointer

Reply:
I am confused. why the requirement on pointer?  timeout validator initializer again [converts](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/openshift-kube-apiserver/authentication/oauth/timeoutvalidator.go#L53) int32 to time.Duration again. We can use `time.Duration` to keep the field consistent as the value propagates from OAuth config to timeout validator.

Reply:
@vareti related and important: https://github.com/openshift/api/pull/658/files#r446205376

Reply:
> I am confused. why the requirement on pointer?
+ as explained in the lib-go PR, you need to have a sense of "unset/set to 0/set to x" states due to https://github.com/openshift/origin/blob/295a35c33057749cc7bdee1f2d33ddaed5b77d60/vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/patch_authenticator.go#L60

Reply:
> related and important: https://github.com/openshift/api/pull/658/files#r446205376

@sttts updated the type to pointer type.

> you need to have a sense of "unset/set to 0/set to x" states

@stlaz I see what you are saying. But I think it can as well achieve similar behavior by not using a pointer. Anyways, I updated the type to pointers, so we should be good for now.